build-system/tasks/e2e/describes-e2e.js

```diff
@@ -266,11 +323,29 @@ class AmpPageFixture {
     const {controller} = env;
     if (controller) {
       await controller.switchToParent();
+      await controller.dispose();
     }
-    if (this.driver_) {
-      await this.driver_.quit();
-    }
-    this.driver_ = null;
+  }
+}
+
+/**
+ * Get the controller object for the configured engine.
+ * @param {!DescribesConfigDef} describesConfig
+ */
+async function getController(describesConfig) {
+  const {
```

Comment:
I would suggest one of two patterns here:

1. Move this to function arguments and default there:

```javascript
async function getController({
  engine = 'selenium',
  headless = false.
}) {
  ...
}
```

This is nice in that the defaulting behavior is now in the function signature. It ties the defaulting behavior more clearly with the function. I'm not sure anything does it currently, but in the future, tools that auto generate function snippets on hover (e.g. Visual Studio Code) could include the default values in the description.

2. Make the defaults explicit an an earlier stage. In this case, I think this approach would be better. It doesn't feel like there is anything particular to this function that it makes it the right place to determine the defaults.

Reply:
I was weighing that decision while I was developing it and I agree that hoisting the defaults to that level will improve the clarity